vendor/bat-native-ads/src/bat/ads/internal/processors/contextual/text_embedding/text_embedding_html_events_database_table.cc

```diff
@@ -24,6 +27,29 @@ namespace {
 
 constexpr char kTableName[] = "text_embedding_html_events";
 
+std::vector<float> ConvertStringToVector(std::string string) {
+  const std::vector<std::string> vector_string = base::SplitString(
+      string, " ", base::TRIM_WHITESPACE, base::SPLIT_WANT_NONEMPTY);
+  std::vector<float> vector;
+  for (const std::string& element_string : vector_string) {
+    double element;
+    base::StringToDouble(element_string, &element);
+    vector.push_back(element);
+  }
+
+  return vector;
+}
+
+std::string ConvertVectorToString(std::vector<float> vector) {
```

Comment:
This seems like it should be defined at a higher level so other functions can use it

Reply:
`JoinFloatAsString` / `SplitStringAsFloat`

Reply:
@tmancey where would the best place be to add this function? I see we have some string util in `src/bat/ads/internal/base/`

Reply:
Please see `vendor/bat-native-ads/src/bat/ads/internal/base/README.md` as I am not sure this would be used by others outside of ml/text embeddings?

Reply:
I moved it to `internal/base/string`, it's general purpose and since it's used in different places between serving and processing thought it could fit there. Otherwise we can have an embedding utility like we have for segments. Wdyt?